library/bigip_ssl_certificate.py

```diff
@@ -0,0 +1,449 @@
+#!/usr/bin/python
+
+
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+# TODO: pkcs12 support when f5-sdk adds it, or convert locally to pem?
+#       Would be nice if f5-sdk file upload could take a string like soap
+#       cert/key import, then we could convert in memory and import
```

Comment:
inclusion of tm.sys.crypto.pkcs12 in the sdk would probably be the most
useful here.

My original through here was to convert PKCS12 to PEM in memory with
OpenSSL and then upload without having to write out a file.  After
discovering the native REST api has pkcs12 functions, this is completely
unnecessary if the f5-sdk exposes that api.

On Mon, Jun 13, 2016 at 12:19 PM, Tim Rupp <notifications@github.com> wrote:

> In library/bigip_ssl_certificate.py
> <https://github.com/F5Networks/f5-ansible/pull/18#discussion_r66822124>:
>
> > +# Ansible is free software: you can redistribute it and/or modify
> > +# it under the terms of the GNU General Public License as published by
> > +# the Free Software Foundation, either version 3 of the License, or
> > +# (at your option) any later version.
> > +#
> > +# Ansible is distributed in the hope that it will be useful,
> > +# but WITHOUT ANY WARRANTY; without even the implied warranty of
> > +# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
> > +# GNU General Public License for more details.
> > +#
> > +# You should have received a copy of the GNU General Public License
> > +# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
> > +#
> > +# TODO: pkcs12 support when f5-sdk adds it, or convert locally to pem?
> > +#       Would be nice if f5-sdk file upload could take a string like soap
> > +#       cert/key import, then we could convert in memory and import
>
> I will speak with Za about this, as I suspect it's within the realm of
> possibility to add
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/F5Networks/f5-ansible/pull/18/files/b156b3c9d2e8dd85246b216eb7c28aeda45f7a6b#r66822124>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALqwBzn61w3JSy2veW6u_VbbIPYLVGLkks5qLYMagaJpZM4I0cLm>
> .
>